File: javascript/bob/bob.js
var Bob = function () { };

Bob.prototype.hey = function (input) {
  input = input.trim()
  if (!input) {
    return "Fine. Be that way!";
  }
  if (input === input.toUpperCase() && input != input.toLowerCase()) {
    return "Whoa, chill out!";
  }
  if (input.slice(-1) === "?") {
    return "Sure.";
  }
  return "Whatever.";
};

module.exports = Bob;

File: go/bob/bob.go
package bob

import "strings"

const testVersion = 3

// Hey returns Bobs response to the input
func Hey(input string) string {
	input = strings.TrimSpace(input)
	if input == "" {
		return "Fine. Be that way!"
	}
	if strings.ToUpper(input) == input && strings.ToLower(input) != input {
		return "Whoa, chill out!"
	}
	if strings.HasSuffix(input, "?") {
		return "Sure."
	}
	return "Whatever."
}
